hongyeollee/app.js

```diff
@@ -33,21 +33,136 @@ app.get("/ping", (req, res) => {
   res.json({ message: "pong" });
 });
 
-//create user
+//과제 2.create user
 app.post("/signup", async (req, res) => {
-  const { name, password, age, email } = req.body;
+  const { name, password, profileImage, email } = req.body;
 
   await mysqlDatabase.query(
     `INSERT INTO users(
           name,
           password,
-          age,
+          profile_image,
           email
         )VALUES(?, ?, ?, ?);
         `,
-    [name, password, age, email]
+    [name, password, profileImage, email]
   );
-  res.status(201).json({ message: "userCreated ✅" });
+  res.status(201).json({ message: "userCreated 👫" });
+});
+
+//과제 3. create post
+app.post("/post", async (req, res) => {
+  const { title, content, imageUrl } = req.body;
+
+  await mysqlDatabase.query(
+    `INSERT INTO posts(
+          title,
+          content,
+          image_url
+        )VALUES(?, ?, ?);
+        `,
+    [title, content, imageUrl]
+  );
+  res.status(201).json({ message: "postCreated 📝" });
+});
+
+//과제 4. search and get posts
+app.get("/getpost", async (req, res) => {
+  await mysqlDatabase.query(
+    `SELECT
+        u.id AS userId,
+        u.profile_image AS userProfileImage,
+        p.id AS postingId,
+        p.image_url AS postingImageUrl,
+        p.content AS postingContent
+    FROM users u
+    INNER JOIN posts p
+    ON u.id=p.id`,
+    (err, rows) => {
+      res.status(200).json({ data: rows });
+    }
+  );
+});
+
+//과제 5. get post of targeting user
+app.get("/postofuser", async (req, res) => {
+  await mysqlDatabase.query(
+    `SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+        JSON_ARRAYAGG(
+          JSON_OBJECT(
+            'postingId',p.id,
+            'postingImageUrl', p.image_url,
+            'postingContent', p.content)) AS posting
+    FROM posts p
+    INNER  JOIN users u
+    ON u.id=p.id
+    GROUP BY u.id;
+    `,
+    (err, rows) => {
+      res.status(200).json({ data: rows });
+    }
+  );
+});
+
+// 과제 6. update posting content
+app.patch("/updatecontent", async (req, res) => {
```

Comment:
아래와 같이 메소드에 의미가 담겨 있기 때문에 게시글을 업데이트 한다 라는 API 라면 `/post` patch 메소드를 사용에서 그 의미를 명확하게 전달할 수 있습니다! 
```suggestion
app.patch("/post", async (req, res) => {
```